osu.Game/Skinning/LegacySkin.cs

```diff
@@ -331,6 +336,8 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                     switch (target.Target)
                     {
                         case SkinnableTarget.MainHUDComponents:
+                            if (!this.HasFont(LegacyFont.Score) && !AllowDefaultHUDComponentsFallback)
+                                return null;
```

Comment:
I could rewrite this so it doesn't get duplicated in here and in `HUDSkinComponent` case, by doing this instead:
```csharp
                            bool canFallback = AllowDefaultHUDComponentsFallback;

                            var components = new[]
                            {
                                GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.ComboCounter)) ?? (canFallback ? new DefaultComboCounter() : null),
                                GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.ScoreCounter)) ?? (canFallback ? new DefaultScoreCounter() : null),
                                GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.AccuracyCounter)) ?? (canFallback ? new DefaultAccuracyCounter() : null),
                                GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.HealthDisplay)) ?? (canFallback ? new DefaultHealthDisplay() : null),
                            };

                            if (components.Any(d => d == null))
                                return null;

                            var skinnableTargetWrapper = new SkinnableTargetComponentsContainer(container =>
                            {
                                var score = container.OfType<LegacyScoreCounter>().FirstOrDefault();
                                var accuracy = container.OfType<GameplayAccuracyCounter>().FirstOrDefault();

                                if (score != null && accuracy != null)
                                {
                                    accuracy.Y = container.ToLocalSpace(score.ScreenSpaceDrawQuad.BottomRight).Y;
                                }
                            })
                            {
                                Children = components,
                            };

                            return skinnableTargetWrapper;
```

but felt like it's not really worth it compared to just doing the check in here and call it a day, given it won't stay in here for long anyway.